### PR TITLE
completed POM homework

### PR DESCRIPTION
**Pull request by RUSTEMATOR**

### CODE REVIEW

File: tests/first.spec.js [hunk 1]
```diff
@@ -1,357 +1,380 @@
...
-test.describe('Positive Reg form tests', async () => {
 
+test.describe('Positive Reg form tests', async () => {
+  let RegPopUp;
+  let randomEmail;
   test.describe('Reg test', async () => {
+    test.beforeEach('Enter the site', async ({page}) => {
+      const welcomePage = new WelcomePage(page)
+      await welcomePage.navigate();
+      RegPopUp = await welcomePage.header.clickRegButton();
```
Comment (sttaran) [inline tests/first.spec.js:14]: RegPopUp is an insyance, it should be caalled `regPopUp`

File: tests/first.spec.js [hunk 2]
```diff
@@ -1,357 +1,380 @@
...
-        await page.locator('.hero-descriptor_btn.btn.btn-primary').click();
+        const welcomePage = new WelcomePage(page)
+        await welcomePage.navigate();
+        RegPopUp = await welcomePage.header.clickRegButton();
+        
       })
 
       test('Check if reg form opens', async ({page}) => {
-        await expect(page.locator('app-signup-modal')).toBeVisible();
+        await expect(RegPopUp.regPopUpModal).toBeVisible();
```
Comment (sttaran) [inline tests/first.spec.js:40]: what is RegPopUp.regPopUpModal?

File: tests/first.spec.js [hunk 3]
```diff
@@ -1,357 +1,380 @@
...
+      await RegPopUp.fillIn({lastName: 'Krako'})
+      await RegPopUp.lastNameField.blur();
+      await expect(RegPopUp.lastNameField).toHaveValue('Krako');
     })
 
     test('Check if email field accepts info', async ({page}) => {
-      await page.locator('#signupEmail').fill('rako@roflan.com');
-      await page.locator('#signupEmail').blur();
-      await expect(page.locator('#signupEmail')).toHaveValue('rako@roflan.com');
+      await RegPopUp.fillIn({email: 'rako@roflan.com'})
```
Comment (sttaran) [inline tests/first.spec.js:92]: save email to a variable and reuse it

Comment (sttaran) [reply tests/first.spec.js:92]: same for other tests

File: src/pageObjects/BasePage.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import {expect} from '@playwright/test';
+import Header from '../components/Header';
+
+
+export default class BasePage{
+    constructor(page, url, WaitPageLocator){
```
Comment (sttaran) [inline src/pageObjects/BasePage.js:6]: WaitPageLocator is an instance of Locator

should be `waitPageLocator`

File: src/pageObjects/welcomePage/WelcomePage.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import BasePage from "../BasePage";
+
+
+export default class WelcomePage extends BasePage  {
+
+    constructor(page) {
+        super(page, '/', page.locator('.header-link.-guest'));
+
+        this.signInButtonText = page.locator('Sign In')
```
Comment (sttaran) [inline src/pageObjects/welcomePage/WelcomePage.js:9]: why?

wrong selector and strange name + this is unused

File: src/pageObjects/welcomePage/WelcomePage.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+        this.signUpButton = page.locator('.hero-descriptor_btn.btn.btn-primary')
```
Comment (sttaran) [inline src/pageObjects/welcomePage/WelcomePage.js:10]: `.hero-descriptor_btn` would be enough

File: src/pageObjects/welcomePage/components/RegPopUp.js [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+
+        this._nameFieldLengthErr = page.locator('div.invalid-feedback > p').filter({hasText: 'Name has to be from 2 to 20 characters long'})
+        this._lastNameFieldLengthErr = page.locator('div.invalid-feedback > p').filter({hasText: 'Last name has to be from 2 to 20 characters long'})
+        this._passwordFieldLengthErr = page.locator('div.invalid-feedback > p').filter({hasText: 'Password has to be from 8 to 15 characters long and contain at least one integer, one capital, and one small letter'})
+
+        this._emailFieldViolationErr = page.locator('div.invalid-feedback > p').filter({hasText: 'Email is incorrect'})
+        this._nameFieldViolationErr = page.locator('div.invalid-feedback > p').filter({hasText: 'Name invalid'})
+        this._lastNameFieldViolationErr = page.locator('div.invalid-feedback > p').filter({hasText: 'Name invalid'})
+
+        this._passDontMatchErr = page.locator('div.invalid-feedback > p').filter({hasText: 'Passwords do not match'})
```
Comment (sttaran) [inline src/pageObjects/welcomePage/components/RegPopUp.js:29]: don't make locators per text

it is the same element but with different content

File: src/pageObjects/garagePage/GaragePage.js [hunk 8]
```diff
@@ -0,0 +1,16 @@
+import BaseComponent from "../welcomePage/components/BaseComponent";
+
+export default class GaragePage extends BaseComponent {
```
Comment (sttaran) [inline src/pageObjects/garagePage/GaragePage.js:3]: `export default class GaragePage extends BasePage`

**Review by sttaran — CHANGES_REQUESTED** [review]

File: tests/first.spec.js (hunk 2 above)
Comment (RUSTEMATOR) [reply tests/first.spec.js:40]: Screenshot at Aug 01 11-34-18

The pop-up itself: <app-signup-modal _nghost-ysg-c48="">

Is there any way to name this element better or this step is to be omitted at all?

File: src/pageObjects/welcomePage/WelcomePage.js (hunk 5 above)
Comment (RUSTEMATOR) [reply src/pageObjects/welcomePage/WelcomePage.js:9]: Don't even remember why I did that. Probably was half asleep

**Pull request closed without merging**